ci/create_package.py

```diff
@@ -0,0 +1,45 @@
+from pathlib import Path
+from shutil import copyfile
+
+Path('my_package').mkdir()
+copyfile('../miniver/miniver/_static_version.py',
+         'my_package/_static_version.py')
+copyfile('../miniver/miniver/_version.py',
+         'my_package/_version.py')
+
+
+# Create the setup file
+with open('../miniver/README.md') as f:
+    for line in f:
+        if line.startswith("# Your project's setup.py"):
+            break
+    with open('setup.py', 'w+') as out:
+        out.write(line)
+        out.write("from setuptools import setup\n")
+        for line in f:
+            if line.startswith('```'):
+                break
+            out.write(line)
+
+# git attributes file
+with open('../miniver/README.md') as f:
+    for line in f:
+        if line.startswith("# Your project's .gitattributes"):
+            break
+    with open('.gitattributes', 'w+') as out:
+        out.write(line)
+        for line in f:
+            if line.startswith('```'):
+                break
+            out.write(line)
+
+with open('../miniver/README.md') as f:
+    for line in f:
+        if line.startswith("# Your package's __init__.py"):
+            break
+    with open('my_package/__init__.py', 'w+') as out:
+        out.write(line)
+        for line in f:
+            if line.startswith('```'):
+                break
+            out.write(line)
```

Comment:
Maybe we could make this block into a function; something like:

```python
def write_snippet_from_readme(start_marker, outfile):
    ...

...

write_snippet_from_readme("# Your project's setup.py", 'setup.py')
write_snippet_from_readme("# Your project's .gitattributes", '.gitattributes')
write_snippet_from_readme("# Your package's __init__.py", 'my_package/__init__.py')
```

What do you think?

Reply:
Right, we could. We would need to find a way to add the line
```
from setuptools import setup
```
to setup.py. I think it is dooable through a second parameter.

Reply:
I would say just add that explicitly:

```
with open('setup.py', 'w+') as f:
    f.writelines('from setuptools import setup')

write_snippet_from_readme("# Your project's setup.py", 'setup.py')
...
```

This special case doesn't seem so special as to deserve an extra parameter IMO